src/pymor/models/loader.py

```diff
@@ -0,0 +1,116 @@
+
+"""Loads models from yml-based configuration files.
+
+The configuration file should be of form:
+
+type: <subclass of pymor.models>
+<operator>
+parameters:
+  <arguments>
+
+With <arguments> being one or many statements of the form
+<argument>:
+  <value>
+OR
+<argument>:
+  - <value>
+  - <value>
+  - ...
+
+With <operator> being one or many statements of the form
+operator:
+  type: <subclass of pymor.operators>
+  <operator>
+  <coefficients>
+OR
+<name>:
+  <matrix>
+OR
+<name>:
+  - <matrix>
+  - <matrix>
+"""
+
+
+def load_module(path):
+    """Creates model from configuration file.
+
+    Args:
+        path (string): Path to the configuration file relative to pymor_source.
+
+    Returns:
+        model: model as loaded from the file.
+    """
+    with open(path, "r") as stream:
+        try:
+            load_dict = yaml.safe_load(stream)
+            print(load_dict)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    # construct the parameter object first
+    parameters = None
+    if ("parameters" in load_dict.keys()):
+        # parameters = Parameters()
+        NotImplemented
+
+    # parse loaded dict, combine it with parameters,
+    # construct the objects and write it in model_parameters dict
+    model_parameters = {}
+    for key, value in load_dict.items():
+        if key == "type":
+            model_parameters["type"] = value
+        # identify operators
+        elif key == "operator" or key == "products" or ".mat" in value:
+            model_parameters[key] = construct_operator(parameters, **value)
+        elif key == "parameters":
+            continue
+        else:
+            raise ValueError(f"The key {key} given is not permitted.")
+    # construct the model
+    return construct_model(**model_parameters)
+
+
+def construct_operator(parameters, param_dict):
+    """Constructs the operator specified in 'type' of the param_dict with the other key-value-pairs.
+
+    The possible operators are all subclasses of pymor.operators.
+    """
+    operator_parameters = {}
+    if param_dict["type"] == "LincombOperator":
+        # parse specific parameters of operator, should be recursive
+        for key, value in param_dict.items():
+            if "diffusion" in value:
+                operator_parameters["diffusion"] = parameters["diffusion"]
+            # and all the other parts of the parameters object
+            elif ".mat" in value:
+                operator_parameters[key] = load_matrix(value)
+
+        # construct operator
+    # elif all the other subclasses of operator
+    else:
+        raise ValueError("The type of operator given is not permitted.")
+
+
+def construct_model(model_parameters):
+    """Constructs the model.model.
+
+    As specified in 'type' of the model_parameters with the other key-value-pairs.
+    The possible models are all subclasses of pymor.models.
+    """
+    if model_parameters["type"] == "StationaryModel":
```

Comment:
I wonder if we can do something more automatic here than manually branching on all model types. 
Once upon a time we had a mechanism that gave a class a list of its subclasses via its [metaclass](https://github.com/pymor/pymor/blob/main/src/pymor/core/base.py#L86). We removed that though since we never actually used it. This might actually be a good enough use-case to bring it back.

Reply:
Another simple possibility to dispatch:

```
import pymor.basic
model_type = getattr(pymor.basic, model_parameters['type']
if not issubclass(model_type, Model):
    raise ValueError('foo')
return model_type.from_yaml(model_parameters)
```

Reply:
Sure. That doesn't work for user models though, or generally models that are not explicitly added in basic

Reply:
The metaclass approach also is somewhat problematic because it requires the module defining the subclass to be imported first. I would also be fine which a simple global dict in the module which maps strings to `Model` subclasses.

Reply:
A user with a new model would then write into that dict?
I think we could also do both approaches, with one being the fallback for the other.

Reply:
Yes and yes.